calyx/src/passes/compile_ref.rs

```diff
@@ -64,6 +65,27 @@ impl Visitor for CompileRef {
             true,
             &mut self.port_names,
         );
+
+        for cell in comp.cells.iter() {
+            let mut new_ports: Vec<RRC<ir::Port>> = Vec::new();
+            if let Some(name) = cell.borrow().type_name() {
+                if self.port_names.contains_key(name) {
+                    for (_, submap) in self.port_names[name].iter() {
```

Comment:
If we do end up deciding that the triply nested Hashmap is needed, let's make this level into a structure and make these two `for` loops into a function that generates all the needed ports from the original data given an `ir::Cell`